### ISSUE TITLE
Fail with higher K-mer sizes

### ISSUE THREAD

**Issue by jgnunes** [maickrau/MBG#3]
Hi, 

When running MBG with K-mer=301, the job finishes successfully. However, I have been running into this error when running MBG with higher K-mer sizes (I've tried both k=401 and k=501):

 ```
MBG bioconda 1.0.5                                                                                                                 
Parameters: k=501,w=250,a=5,u=150,t=16,errormasking=hpc,endkmers=no,blunt=no                                                    
Collecting selected k-mers                                                                                                           
Reading sequences from gbk.HiFiMapped.bam.filtered.fasta                                                                          
136112 total selected k-mers in reads                                                              
36605 distinct selected k-mers in reads                                                                                      
Unitigifying                                                                                                                       
MBG: src/MBG.cpp:687: UnitigGraph getUnitigGraph(const HashList&, size_t): Assertion result.unitigs.back().back() == reverse(bwEdges[0])' failed.                                                                                
/tmp/1627073349.984834.shell: line 1: 36524 Aborted                 MBG -i gbk.HiFiMapped.bam.filtered.fasta -o mbg_assembly -k 501 -w 250 -a 5 -u 150 -t 16                                                                                                          
 ```

Any clue on what may be causing this? 
Let me know if you need any additional information. 

Best,

**Comment by maickrau** [comment]
Hi, could you please upload the reads where this issue happens?

**Comment by jgnunes** [comment]
Yes sure, here it goes:

[gbk.HiFiMapped.bam.filtered.fasta.gz](https://github.com/maickrau/MBG/files/6881478/gbk.HiFiMapped.bam.filtered.fasta.gz)

**Comment by maickrau** [comment]
This is fixed in commit 3a3c79c. The crash was caused by this strange barbell shaped contig (maybe some kind of palindromic repeat?):

![barbell_contig](https://user-images.githubusercontent.com/1543674/127498870-3cc92b88-3b70-4a9a-8dbc-4666e257f5a0.png)

**Comment by jgnunes** [comment]
Thanks, that solved the problem!
